kf/nfconfig.go

```diff
@@ -937,3 +938,219 @@ func getHostInterfaces() (map[string]bool, error) {
 	}
 	return hostIfaces, nil
 }
+
+//  AddProgram : AddProgram will add given programs on given interface
+func (c *NFConfigs) AddProgram(ifaceName, hostName string, bpfProgs *models.BPFPrograms) error {
+
+	if hostName != c.hostName {
+		errOut := fmt.Errorf("provided bpf programs do not belong to this host")
+		log.Error().Err(errOut)
+		return errOut
+	}
+
+	if ifaceName == "" || bpfProgs == nil {
+		errOut := fmt.Errorf("iface name or bpf programs are empty")
+		log.Error().Err(errOut)
+		return errOut
+	}
+
+	if _, ok := c.hostInterfaces[ifaceName]; !ok {
+		errOut := fmt.Errorf("%s interface name not found in the host", ifaceName)
+		log.Error().Err(errOut)
+		return errOut
+	}
+
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	for _, bpfProg := range bpfProgs.XDPIngress {
+		if c.IngressXDPBpfs[ifaceName] == nil {
+			if bpfProg.AdminStatus == models.Enabled {
+				c.IngressXDPBpfs[ifaceName] = list.New()
+				if err := c.VerifyAndStartXDPRootProgram(ifaceName, models.XDPIngressType); err != nil {
+					c.IngressXDPBpfs[ifaceName] = nil
+					return fmt.Errorf("failed to chain XDP BPF programs: %w", err)
+				}
+				log.Info().Msgf("Push Back and Start XDP program : %s seq_id : %d", bpfProg.Name, bpfProg.SeqID)
+				if err := c.PushBackAndStartBPF(bpfProg, ifaceName, models.XDPIngressType); err != nil {
+					return fmt.Errorf("failed to pushbackandstart BPF Program: %w", err)
+				}
+			}
+		} else if err := c.InsertAndStartBPFProgram(bpfProg, ifaceName, models.XDPIngressType); err != nil {
+			return fmt.Errorf("failed to InsertandStartBPFProgram xdp BPF Program: %w", err)
+		}
+	}
+
+	for _, bpfProg := range bpfProgs.TCIngress {
+		if c.IngressTCBpfs[ifaceName] == nil {
+			if bpfProg.AdminStatus == models.Enabled {
+				c.IngressTCBpfs[ifaceName] = list.New()
+				if err := c.VerifyAndStartTCRootProgram(ifaceName, models.IngressType); err != nil {
+					c.IngressTCBpfs[ifaceName] = nil
+					return fmt.Errorf("failed to chain ingress tc bpf programs: %w", err)
+				}
+				if err := c.PushBackAndStartBPF(bpfProg, ifaceName, models.IngressType); err != nil {
+					return fmt.Errorf("failed to pushbackandstart BPF Program: %w", err)
+				}
+			}
+		} else if err := c.InsertAndStartBPFProgram(bpfProg, ifaceName, models.IngressType); err != nil {
```

Comment:
Why does "InsertAndStartBPFProgram" end in Program but "PushBackAndStartBPF" does not?   This seems inconsistent.  I suspect the latter should be "PushBackAndStartBPFProgram".